testnet/stacks-node/src/neon_node.rs

```diff
@@ -138,7 +138,7 @@ fn inner_process_tenure(
 
     // todo(ludo): yikes but good enough in the context of helium:
     // we only expect 1 block.
-    let processed_block = match processed_blocks.get(0) {
+    let processed_block = match processed_blocks.last() {
```

Comment:
This snippet should probably do just two things:

1. Warn if processed_blocks.len() == 0,
2. Iterate over all the processed_blocks, and dispatch the events from them.

Then we should just return a `Result<(), ChainstateError>` type from this function, and then at the caller, rather than use the `stacks_header` return value, just use the miner's own tracked header information for the advertize+broadcast (as implemented, there could theoretically be cases where a miner push broadcasts a block that is "higher" than the block it just mined).

Reply:
Addressed in 5c39c2f